### PR TITLE
fix: profile status buttons UI [AR-1584]

### PR DESCRIPTION
**Pull request by Garzas**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1584" title="AR-1584" target="_blank">BugAR-1584</a>  Change status button visuals are broken
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

_Briefly describe the issue you have solved or implemented with this pull request. If the PR contains multiple issues, use a bullet list._

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-569/1/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3320557)(3320557)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3320557)(3320557)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileScreen.kt [hunk 1]
```diff
@@ -186,71 +189,77 @@ private fun CurrentSelfUserStatus(
...
-                    modifier = Modifier.padding(end = dimensions().spacing4x).testTag("Away")
-                )
-            }
+            currentStatus = userStatus,
         )
-        WireSecondaryButton(
-            onClick = { onStatusClicked(UserStatus.NONE) },
+        ProfileStatusButton(
+            userStatus = UserStatus.NONE,
+            modifier = Modifier.weight(weight = 3F),
```
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileScreen.kt:219]: I'm wondering if this fixed values `3F`, `4F`, `6F`, will resist all major cases 🤔  
Even tough, we currently not support landscape mode, will be a good time to do a little test before committing to these values. What do you think ?

**Review by yamilmedina — COMMENTED** [review]
LGTM in general, I left a small comment about the possibility to test landscape mode (not supported yet) just see how these values behave, What do you think ?

File: app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileScreen.kt (hunk 1 above)
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileScreen.kt:219]: Unfortunately not all issues. Does in future we will have other languages? I Can make button weight dependable on text size

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-569/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3321229)(3321229)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3321229)(3321229)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileScreen.kt (hunk 1 above)
Comment (yamilmedina) [reply app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileScreen.kt:219]: Yes! In fact we currently support l10n 😅  we'll have other languages on those buttons

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-569/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3336961)(3336961)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3336961)(3336961)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/common/UserProfileAvatar.kt [hunk 2]
```diff
@@ -69,6 +68,25 @@ fun UserProfileAvatar(
     }
 }
 
+/**
+ * Workaround to have profile avatar available for preview
+ * @see [painter] https://developer.android.com/jetpack/compose/tooling
```
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/ui/common/UserProfileAvatar.kt:73]: Cool trick ! 👍  🚀

**Review by yamilmedina — APPROVED** [review]
Great work @Garzas ! Nice to have these dynamic weights 🚀 🥇

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-569/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3337808)(3337808)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3337808)(3337808)-dev-release.apk)

**Pull request merged**